auto_dev/workflow_manager.py

```diff
@@ -257,6 +257,62 @@ def load_yaml(file_path: str):
         with open(file_path, encoding="utf-8") as file:
             return yaml.safe_load(file)
 
+    @staticmethod
+    def load_custom_workflow(workflow_file: str, params: str) -> "WorkflowManager":
+        """Get a workflow with custom parameters.
+
+        Args:
+        ----
+            workflow_file (str): Path to the workflow YAML file.
+            params (str): A string containing key-value pairs for placeholders in format "key1=value1,key2=value2".
+
+        Returns:
+        -------
+            WorkflowManager: A WorkflowManager instance with the configured workflow.
+
+        Example:
+        -------
+           wf = WorkflowManager.load_custom_workflow("workflow.yaml", "public_id=author/name,param1=value1")
+
+        """
+        parameters = dict(param.split("=") for param in params.split(",") if param)
+        # Check if publicId is in parameters and split it into author and name
+        if "public_id" in parameters:
+            author, name = parameters["public_id"].split("/")
+            parameters["author"] = author
+            parameters["agent_name"] = name
+
+        workflow_yaml = WorkflowManager.load_yaml(workflow_file)
+
+        # Replace placeholders in the loaded YAML data
+        def replace_placeholders(obj):
+            if isinstance(obj, dict):
```

Comment:
I can see this having issues with Json and yaml inputs.

Effectively json and yaml have different key_pair settings,
So we would want to just accept kv pairs at this point and offload the io from disk/memory to somewhere else